main/src/ca/uwaterloo/flix/api/lsp/provider/completion/CompletionUtils.scala

```diff
@@ -228,4 +229,33 @@ object CompletionUtils {
   private def splitByCamelCase(input: String): List[String] = {
     input.split("(?=[A-Z])").toList
   }
+
+  /**
+    * Updates the given `qn` based on the character immediately following the location.
+    * If the character is a dot, we move the ident to the namespace.
+    * Otherwise, we will keep the `qn` as it is.
+    *
+    * Example:
+    *   - Source "AA.BB", QName(["AA"], "BB") -> QName(["AA"], "BB")
+    *   - Source "AA.BB.", QName(["AA"], "BB") -> QName(["AA", "BB"], "")
+    */
+  def updateQNameBasedOnDot(qn: QName, loc: SourceLocation): QName = {
```

Comment:
As a general rule, we should __never__ construct QNames as they come from the AST. 

Why do you need to do that?

Reply:
OK. I will use a List[String] for the namespace and a String for the ident.